actions/remote.go

```diff
@@ -0,0 +1,168 @@
+package actions
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/qri-io/dag"
+	"github.com/qri-io/dag/dsync"
+	"github.com/qri-io/qri/p2p"
+	"github.com/qri-io/qri/repo"
+	"github.com/qri-io/qri/repo/profile"
+)
+
+// DsyncStartPush posts the dataset's dag.Info to the remote to get a sessionID and diff
+func DsyncStartPush(node *p2p.QriNode, dagInfo *dag.Info, location string, ref *repo.DatasetRef) (string, *dag.Manifest, error) {
+	node.LocalStreams.PrintErr("posting to /dsync/push...\n")
```

Comment:
Great to see use of `PrintErr` here despite us not having a proper set of guidelines on use. I'm also seeing fewer / no `fmt.PrintX`. tyvm